build-logic/publication/src/main/kotlin/convention.publish-base.gradle.kts

```diff
@@ -9,6 +11,16 @@ version = providers.gradleProperty("projectVersion")
     .forUseAtConfigurationTime()
     .get()
 
+tasks.withType(Jar::class.java) {
```

Comment:
you can use reified generic version: `tasks.withType<Jar> {`

Reply:
Fixed